book-content/chapters/11-annotations-and-assertions.md

```diff
@@ -0,0 +1,1196 @@
+
+But it's useful in cases where we truly know more than TypeScript can about the behavior of our code.
+
+#### An Alternative Syntax
+
+As an alternative to `as`, you can prefix the value with the type wrapped in angle brackets:
+
+```typescript
+const id = <string>searchParams.get("id");
+```
+
+This is less common than `as`, but behaves exactly the same way. `as` is more common, so it's better to use that.
+
+#### The Limits of `as`
+
+`as` has some limits on how it can be used. It can't be used to convert between unrelated types.
+
+Consider this example where `as` is used to assert that a string should be treated as a number:
+
+```ts twoslash
+// @errors: 2352
+const albumSales = "Heroes" as number;
+```
+
+TypeScript realizes that even though we're using `as`, we might have made a mistake. The error message is telling us that a string and a number don't share any common properties, but if we really want to go through with it, we could double up on the `as` assertions to first assert the string as `unknown` and then as a `number`:
+
+```tsx
+const albumSales = "Heroes" as unknown as number; // no error
+```
+
+When using `as` to assert as `unknown as number`, the red squiggly line goes away but that doesn't mean the operation is safe. There's just no way to convert `"Heroes"` into a number that would make sense.
+
+The same behavior applies to other types as well.
+
+In this example, an `Album` interface and a `SalesData` interface don't share any common properties:
+
+```ts twoslash
+// @errors: 2352
+interface Album {
+  title: string;
+  artist: string;
+  releaseYear: number;
+}
+
+interface SalesData {
+  sales: number;
+  certification: string;
+}
+
+const paulsBoutique: Album = {
+  title: "Paul's Boutique",
+  artist: "Beastie Boys",
+  releaseYear: 1989,
+};
+
+const paulsBoutiqueSales = paulsBoutique as SalesData;
+```
+
+Again, TypeScript shows us the warning about the lack of common properties.
+
+So, `as` does have some built-in safeguards. But by using `as unknown as X`, you can easily bypass them. And because `as` does nothing at runtime, it's a convenient way to lie to TypeScript about the type of a value.
+
+### The Non-null Assertion
+
+Another assertion we can use is the non-null assertion, which is specified by using the `!` operator. This provides a quick way to tell TypeScript that a value is not `null` or `undefined`.
+
+Heading back to our `searchParams` example from earlier, we can use the non-null assertion to tell TypeScript that `id` will never be `null`:
+
+```typescript
+const searchParams = new URLSearchParams(window.location.search);
+
+const id = searchParams.get("id")!;
+```
+
+This forces TypeScript to treat `id` as a string, even though it could be `null` at runtime. It's the equivalent of using `as string`, but is a little more convenient.
```

Comment:
```suggestion
This forces TypeScript to treat `id` as a string, even though it could be `null` at runtime. It's the equivalent of using `as string` in this example, but is a little more convenient.
```

It's more like an equivalent of `as NonNullable<SomeNullableType>`/`as NonNullable<typeof foo>`